akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala

```diff
@@ -803,8 +812,47 @@ trait FlowOps[+Out, +Mat] {
    * '''Cancels when''' downstream cancels and substreams cancel
    *
    */
-  def splitWhen[U >: Out](p: Out ⇒ Boolean): Repr[Source[U, Unit], Mat] =
-    andThen(SplitWhen(p.asInstanceOf[Any ⇒ Boolean]))
+  def splitWhen[U >: Out](p: Out ⇒ Boolean): Repr[Out, Mat]#Repr[Source[U, Unit], Mat] = {
+    val f = p.asInstanceOf[Any ⇒ Boolean]
+    withAttributes(name("splitWhen")).andThen(Split(el ⇒ if (f(el)) SplitBefore else Continue))
+  }
```

Comment:
And on https://github.com/akka/akka/pull/17196/files#diff-d25227230bef7fd44d678521a8f07f1cR854 as well